sky/serve/core.py

```diff
@@ -60,9 +62,18 @@ def up(
         with ux_utils.print_exception_no_traceback():
             raise RuntimeError('Service section not found.')
 
-    requested_cloud: Optional['clouds.Cloud'] = None
-    service_port: Optional[int] = None
     for requested_resources in task.resources:
+        first_resource_dict = list(task.resources)[0].to_yaml_config()
```

Comment:
nit: move this outside of the loop to avoid multiple constructions. Also check the length of `task.resources`